refactor: added absolute import paths to unit tests

timDeHof opened the pull request:
# Description

<!--Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

This Pull Request includes refactoring changes to update import paths and Jest configuration for unit tests. The changes improve the overall structure of the codebase, making it easier to maintain and test.

Key changes include:

- Updated import paths to use the "@" alias for better readability and to avoid relative path issues in unit tests
- Modified Jest configuration in the `package.json` to align with the new import paths and improve test execution
- Updated `tsconfig.json` to reflect changes in import paths

These changes enhance code organization, improve navigation through the codebase, and ensure proper test execution with new structure.

## Issue link

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update


# How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->
Verify that the changes work by:

1. pull the latest changes
2. Run `yarn test` or `yarn test:docker` (if you use docker)
3. Check that the `afterall()` in the healthcheck test by running `yarn studio:test` and see that healthcheck table is empty 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have updated the change log

cherylli commented on CHANGELOG.md:
```diff
@@ -18,6 +18,7 @@ Another example [here](https://co-pilot.dev/changelog)
 
 - Update PostgreSQL image to timescale/timescaledb-ha:pg16 to match railway postgres image ([#210](https://github.com/chingu-x/chingu-dashboard-be/pull/210))
 - refactored e2e tests to use absolute import paths ([#207](https://github.com/chingu-x/chingu-dashboard-be/pull/207))
+- refactored unit tests to use absolute import paths ([#215](https://github.com/chingu-x/chingu-dashboard-be/pull/215))
```
ideally we want to have these in order of completion / merge, since #206 is merged, we should move this below it

cherylli reviewed (changes requested):
Everything works fine but I see some controllers file still has relative imports, e.g. `auth.controller.ts`, were they supposed to be in last PR or next?

![image](https://github.com/user-attachments/assets/f79c0082-2629-4078-8e57-9e76b448ff5a)

timDeHof commented:
> Everything works fine but I see some controllers file still has relative imports, e.g. `auth.controller.ts`, were they supposed to be in last PR or next?
> 
> ![image](https://private-user-images.githubusercontent.com/6191116/376083542-f79c0082-2629-4078-8e57-9e76b448ff5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4Njk3NjYsIm5iZiI6MTcyODg2OTQ2NiwicGF0aCI6Ii82MTkxMTE2LzM3NjA4MzU0Mi1mNzljMDA4Mi0yNjI5LTQwNzgtOGU1Ny05ZTc2YjQ0OGZmNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTRUMDEzMTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2U0Y2UzNTAwZjgzMGZlYmE2NzA5ODk5NmZiZGViZGQyYWY4MTk5NzQ5ZDFhOTM0ZDk2YTBlNmQwMzEwZGJlMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.h0fdT8hm7tFcyhDtaJcIYTIF-A8-UNIYdIcoFIn4Nu8)

oops, I thought you only wanted the tests...I'll go through and change the relative imports to absolutes now.

cherylli commented:
> > Everything works fine but I see some controllers file still has relative imports, e.g. `auth.controller.ts`, were they supposed to be in last PR or next?
> > ![image](https://private-user-images.githubusercontent.com/6191116/376083542-f79c0082-2629-4078-8e57-9e76b448ff5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4Njk3NjYsIm5iZiI6MTcyODg2OTQ2NiwicGF0aCI6Ii82MTkxMTE2LzM3NjA4MzU0Mi1mNzljMDA4Mi0yNjI5LTQwNzgtOGU1Ny05ZTc2YjQ0OGZmNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTRUMDEzMTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2U0Y2UzNTAwZjgzMGZlYmE2NzA5ODk5NmZiZGViZGQyYWY4MTk5NzQ5ZDFhOTM0ZDk2YTBlNmQwMzEwZGJlMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.h0fdT8hm7tFcyhDtaJcIYTIF-A8-UNIYdIcoFIn4Nu8)
> 
> oops, I thought you only wanted the tests...I'll go through and change the relative imports to absolutes now.

Thanks, for some reason I thought all the other ones were changed last time, and only tests left

timDeHof commented:
I went back to the how the old testregex was to fix how the units command was running the e2e tests after the unit tests. Also I corrected the absolute path in the jest section in the package.json.

cherylli commented:
Updated reviewer to @JoshuaHinman

cherylli commented:
I'm getting this when I run prisma db seed , do I need to do extra setup?
![image](https://github.com/user-attachments/assets/2c5bf15e-7d1e-439c-9920-2a1043add9de)

timDeHof commented:
ok, I fixed it by installing `tsconfig-path` package to help with the path-mapping and refactored the prisma seed script to include it to correctly resolve the paths.

cherylli reviewed (approved):
Everything works fine (tests, seed), whats the difference between `@/prisma` and `@Prisma`? looks like we need to use `@Prisma` in e2e tests

timDeHof commented:
> Everything works fine (tests, seed), whats the difference between `@/prisma` and `@Prisma`? looks like we need to use `@Prisma` in e2e tests

I opted for distinct absolute import paths for the prisma directories to avoid potential conflicts when importing from `@prisma/*` modules, as our setup requires referencing two separate prisma directories. One in the project root used for seeding and database connections. The other one in the `src` folder for the services and controllers.

JoshuaHinman reviewed (approved):
All tests passed and health check table is empty - looks good!
